AppController/lib/cassandra_helper.rb

```diff
@@ -174,12 +175,19 @@ def needed_for_quorum(total_nodes, replication)
   total_nodes - can_fail
 end
 
-# Returns the number of nodes in 'Up Normal' state.
+# Returns an array of nodes in 'Up Normal' state.
 def nodes_ready
-  output = `"#{NODETOOL}" status`
-  nodes_ready = 0
+  output = `"#{NODETOOL}" gossipinfo`
+  return [] unless $?.exitstatus == 0
+
+  live_nodes = Set[]
+  current_node = nil
   output.split("\n").each { |line|
```

Comment:
would you mind putting a comment about the expected output of this command? in case something breaks in the future it'd be easier to find the problem.